web/source/osk/visualKeyboard.ts

```diff
@@ -957,7 +880,13 @@ namespace com.keyman.osk {
         console.error("OSK key with ID '" + e.id + "', keyID '" + e.keyId + "' missing needed specification");
         return null;
       }
+      
+      // Return the event object.
+      return this.keyEventFromSpec(keySpec, touch);
+    }
 
+    keyEventFromSpec(keySpec: keyboards.ActiveKey, touch?: Touch) {
+      let core = com.keyman.singleton.core; // only singleton-based ref currently needed here.
```

Comment:
Split in order to facilitate the `KeyEvent` typed-`Promise`s of the `RealizedGesture` classes.